src/map/pc.cpp

```diff
@@ -4777,6 +4778,82 @@ bool pc_skill(struct map_session_data* sd, uint16 skill_id, int level, enum e_ad
 	}
 	return true;
 }
+
+bool pc_skill_plagiarism(struct map_session_data *sd, uint16 skill_id, uint16 skill_lv)
+{
+	if (sd != nullptr) {
+		uint16 idx = 0;
+
+		int type = skill_isCopyable(sd, skill_id);
+		skill_id = skill_dummy2skill_id(skill_id);
+		skill_lv = cap_value(skill_lv, 1, skill_get_max(skill_id));
+
+		//Use skill index, avoiding out-of-bound array [Cydh]
+		if (!(idx = skill_get_index(skill_id))) {
+			ShowWarning("pc_skill_plagiarism: invalid skill idx %d for skill %d.\n", idx, skill_id);
+			return false;
+		}
+
+		if (type == 1) {
+			pc_skill_plagiarism_reset(sd, type);
+
+			sd->cloneskill_idx = idx;
+			pc_setglobalreg(sd, add_str(SKILL_VAR_PLAGIARISM), skill_id);
+			pc_setglobalreg(sd, add_str(SKILL_VAR_PLAGIARISM_LV), skill_lv);
+		}
+		else if (type == 2) {
+			pc_skill_plagiarism_reset(sd, type);
+
+			sd->reproduceskill_idx = idx;
+			pc_setglobalreg(sd, add_str(SKILL_VAR_REPRODUCE), skill_id);
+			pc_setglobalreg(sd, add_str(SKILL_VAR_REPRODUCE_LV), skill_lv);
+		}
+
+		sd->status.skill[idx].id = skill_id;
+		sd->status.skill[idx].lv = skill_lv;
+		sd->status.skill[idx].flag = SKILL_FLAG_PLAGIARIZED;
+		clif_addskill(sd, skill_id);
+	}
+
+	return true;
+}
+
+bool pc_skill_plagiarism_reset(struct map_session_data *sd, uint8 type)
```

Comment:
```suggestion
/**
 * Clear plagiarized skills from a player.
 * @param sd: Player
 * @param type: 1 for Plagiarism or 2 for Reproduce
 * @return True on success or false otherwise
 */
bool pc_skill_plagiarism_reset(map_session_data *sd, uint8 type)
```